aspnetcore/grpc/why-migrate-wcf-to-dotnet-grpc.md

```diff
@@ -0,0 +1,69 @@
+
+## Similarity to WCF
+
+Although the implementation and approach are different for gRPC, the experience of developing and consuming services with gRPC should be intuitive for WCF developers. WCF and gRPC are RPC (remote procedure call) frameworks with the same goal: 
+
+* Make it possible to code as though the client and server are on the same platform.
+* Provide a simplified portable networking API.
+
+Both platforms share the requirement of declaring and implementing an interface, although the process for declaring the interface is different. The many types of RPC calls that gRPC supports map well to the bindings available to WCF services. For more information and examples, see [Migrate a WCF solution to gRPC](/dotnet/architecture/grpc-for-wcf-developers/migrate-wcf-to-grpc)
+
+## Benefits of gRPC
+
+gRPC provides a better framework than other approaches for the following reasons.
+
+### Performance
+
+gRPC uses HTTP/2.  HTTP/2 compared to HTTP/1.1, HTTP/2:
+
+* Is a smaller, faster binary protocol.
+* Is more efficient for computers to parse.
+* Supports multiplexing requests over a single connection. Multiplexing enables multiple requests to be sent over one connection without requests blocking each other. In HTTP/1.1, the blocking is known as "head-of-line (HOL) blocking."
+
+gRPC uses the Protobuf, an efficient binary format, to serialize messages. Protobuf messages are:
+* Small in size.
+* Use less bandwidth than text-based formats. 
+
+gRPC is a good solution to use for mobile devices and over slower networks where bandwidth is important.
+
+### Interoperability
+
+There are gRPC tools and libraries for all major programming languages and platforms, including .NET, Java, Python, Go, C++, Node.js, Swift, Dart, Ruby, and PHP. Thanks to the Protocol Buffers binary wire format and the efficient code generation for each platform, developers can build cross-platform performant apps.
+
+### Usability and productivity
+
+gRPC is a comprehensive RPC solution. It works consistently across multiple languages and platforms. It also provides excellent tooling, with much of the boilerplate code automatically generated. Like WCF, gRPC automatically generates messages and a strongly typed client. Developer time is freed up to focus on business logic.
+
+### Streaming
+
+gRPC has full bidirectional streaming, which provides similar functionality to WCF's full duplex services. gRPC streaming can operate over regular internet connections, load balancers, and service meshes.
+
+### Deadlines, timeouts and cancellation
+
+gRPC allows clients to specify a maximum time for an RPC to finish. If the specified deadline is exceeded, the server can cancel the operation independently of the client. Deadlines and cancellations can be propagated through further gRPC calls to help enforce resource usage limits. Clients can stop operations when a deadline is exceeded, or earlier if necessary. For example,  clients can stop operations because of a user interaction.
```

Comment:
```suggestion
gRPC allows clients to specify a maximum time for an RPC to finish. If the specified deadline is exceeded, the server can cancel the operation independently of the client. Deadlines and cancellations can be propagated through subsequent gRPC calls to help enforce resource usage limits. Clients can stop operations when a deadline is exceeded, or earlier if necessary. For example, clients can stop operations because of a user interaction.
```